### ISSUE TITLE
Downloaded CSV/JSON for a customized table contains all columns

### ISSUE THREAD

**Issue by jgoclawski** [metabase/metabase#3992]
- Your browser and the version: Google Chrome 54
- Your operating system: Ubuntu 16.04
- Your databases: Postgres, H2 (demo)
- Metabase version: 0.21.0
- Metabase hosting environment: Docker
- Metabase internal database: Postgres

*Repeatable steps to reproduce the issue*:
1. Ask any question
2. Display it as table.
3. Click "cog" icon to show preferences for table visualization.
4. A "Customize this table" modal will be shown.
5. Uncheck a few checkboxes in "Fields to include"
6. Click "Done".
7. Observe that the table has only those columns left checked in step 5, e.g "ID", "Total"
8. Click download icon.
9. Select CSV or JSON (doesn't matter)
10. Observe that downloaded file contains all the columns, not only those left checked in step 5.

I'm attaching 2 screenshots.
![metabase_customize_table](https://cloud.githubusercontent.com/assets/1043875/21387523/40650668-c778-11e6-83e5-dde76c80386c.png)
![metabase_download_csv](https://cloud.githubusercontent.com/assets/1043875/21387524/40654682-c778-11e6-9efd-364e187db932.png)

The resulting CSV file starts with:
```
ID,CREATED_AT,PRODUCT_ID,SUBTOTAL,TAX,TOTAL,USER_ID
1,2016-05-28T16:08:40.953Z,141,12.65,0.0,12.65,2
2,2016-08-27T23:41:37.372Z,173,54.8,0.0,54.8,2
3,2016-08-19T22:01:27.966Z,140,29.88,0.0,29.88,2
4,2017-01-19T13:22:38.527Z,196,17.97,0.0,17.97,2
5,2017-03-26T07:12:11.671Z,68,18.1,0.0,18.1,2
```

**Comment by tlrobinson** [comment]
Yeah, this is definitely confusing. The chart settings are all client-side display settings, while the CSV download just downloads the raw query. Same issue with pivoting: #2473

**Comment by tanzoniteblack** [comment]
Do you think this would be solvable if we update the question to write the query such that it only requested from the backend the queries desired to be viewed, and respected the order returned from the backend?

That would potentially solve both this exact problem and another related one I'm having where the order of columns isn't being respected in the download.

If you guys don't have time to do this, but are ok with the approach that I just suggested, then let me know and I'll give it a stab.

**Comment by tlrobinson** [comment]
@tanzoniteblack Yeah I think basically syncing the columns in the display settings with the query would be a good approach, as it would also reduce the amount of data that needs to be fetched.

Related to that I started on a PR awhile ago that allowed you add new columns to a query: #3043

**Comment by Serfenia** [comment]
I really like this enhancement to be added. Thumbs up!

**Comment by csawyerYumaed** [comment]
This seems very much related to https://github.com/metabase/metabase/issues/5295 (XLSX)
and I hope order would be preserved as well.

**Comment by kreno** [comment]
Yes, please.  The export isn't much good to our users if it is different than the settings on the screen.

**Comment by tashiritzy** [comment]
If you use Grouped By option instead of selecting the fields from the check list, it serves some purpose.
